src/components/kalman_component/kalman.sh

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+
+# Functions available in this file include:
+#   - setup_kf
+#   - run_kf
+
+# Description: Setup Kalman filter prereqiuisites
+# Usage:
+#   setup_kf
+setup_kf() {
+    StateVectorFile="${RunDirs}/StateVector.nc"
+
+    # Create a parent directory for the Kalman filter inversions
+    # Include a link to the state vector file for use with run_inversion.sh
+    mkdir -p ${RunDirs}/kf_inversions
+    ln -sf $StateVectorFile ${RunDirs}/kf_inversions/StateVector.nc
+
+    # copy kf notebook to kf_inversions directory
+    cp ${InversionPath}/src/notebooks/kf_notebook.ipynb ${RunDirs}/kf_inversions/
+
+    # Define Kalman filter update periods
+    python ${InversionPath}/src/components/kalman_component/make_periods_csv.py $StartDate $EndDate $UpdateFreqDays $RunDirs; wait
+
+    # Create unit scale factor file
+    python ${InversionPath}/src/components/kalman_component/make_unit_sf.py $StateVectorFile $RunDirs; wait
+
+    # Create directory to archive prior scale factors for each inversion period
+    mkdir -p ${RunDirs}/archive_sf
+
+    # Number of state vector elements
+    nElements=$(ncmax StateVector ${StateVectorFile})
+}
+
+# Description: Run Kalman filter inversions
+# Usage:
+#   run_kf
+run_kf() {
+
+    if ("$DoJacobian" && "$DoInversion" && "$DoPosterior"); then
+
+        # First run the Preview if necessary to get prior emissions
+        # needed for prepare_sf.py
+        if [[ ! -d ${RunDirs}/preview_run/OutputDir ]]; then
+            printf "\nPreview Dir not detected. Running the IMI Preview as a prerequisite for Kalman Mode.\n"
+            run_preview
+        fi
+        # Key directories
+        JacobianRunsDir="${RunDirs}/jacobian_runs"
+        PriorRunDir="${JacobianRunsDir}/${RunName}_0000"
+        PosteriorRunDir="${RunDirs}/posterior_run"
+        InversionDir="${RunDirs}/inversion_template"
+
+        PeriodsFile="${RunDirs}/periods.csv"
+        nPeriods=$(($(wc -l < ${PeriodsFile}) - 1))
+
+        # by default, start with period 1
+        if [[ "x${FirstPeriod}" == "x" ]]; then
```

Comment:
Is this why there is no `FirstPeriod` setting in the default config file? If you don't include it, the IMI knows to start at period 1, and if you want to start at another period then you add the setting?

Reply:
exactly